### DISCUSSION TITLE
Use ci-vite-react only in a section of the page

### DISCUSSION THREAD

**Discussion by sanchawebo** [firtadokei/codeigniter-vitejs#27] [Q&A] [answered]
So I have an existing project in CI4 and want to include react in a section of a page (Which is possible according to the react docs).
I got everything to run and then deactivated VITE_AUTO_INJECTING, as I only want it active on one route and there embed it into my existing page layout. But i can't get it to work.

How do i achieve this?

**Comment by sanchawebo** [comment]
Got it to work like this:
```php
<?php
use Mihatori\CodeigniterVite\Vite;

echo $this->extend('templates/layoutFlaPo');
echo $this->section('content'); ?>

<div class="vite-wrapper">
    <div id="app"></div>
</div>

<?= $this->endSection() ?>

<?php
echo $this->section('pageScripts');
$vite = Vite::tags();
echo $vite['js'] ?? null;
echo $this->endSection() ?>
```

Is this the correct way of embedding vite-react into a page?

**Reply by vago-me** [reply, marked as answer]
use Mihatori\CodeigniterVite\Vite;  $vite = Vite::tags(); $vite['js'] ?? null; 
Should be in the controller, like this.
<?php
namespace App\Controllers;
use Mihatori\CodeigniterVite\Vite;

class Home extends BaseController
{
    public function index()
    {
        $vite = Vite::tags();
        $data['vite'] =  $vite['js'] ?? null;
        return view('Home/welcome_message',$data);
    }
}

and the View 

<?php

<?= $this->extend('templates/layoutFlaPo') ?>
<?= $this->section('content'); ?>

<div class="vite-wrapper">
    <div id="app"></div>
    <?= $vite ?>
</div>

<?= $this->endSection() ?>


in .env

VITE_AUTO_INJECTING=false
